core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -50,12 +50,16 @@ oppia.factory('explorationData', [
     var explorationData = {
       explorationId: explorationId,
       autosaveChangeList: function(changeList, successCallback, errorCallback) {
+        // First save locally to be retrieved later if save is unsuccessful.
+        localSaveService.saveExplorationDraft(explorationId, changeList,
+          draftChangeListId);
         $http.put(explorationDraftAutosaveUrl, {
           change_list: changeList,
           version: explorationData.data.version
         }).then(function(response) {
+          draftChangeListId = response.data.draft_change_list_id;
+          localSaveService.removeExplorationDraft(explorationId);
           if (successCallback) {
-            draftChangeListId = response.data.draft_change_list_id;
             successCallback(response);
           }
         }, function() {
```

Comment:
I think that does explain it, thanks! Do we have plans for adding that modal that displays the potentially-lost changes? If it's part of the milestones, that sounds good -- it's an important use case, I think.